implement --clean flag for build command

appeltel opened the pull request:
Implemented "--clean / -c" flag for build command which removes an existing venv. Currently this just performs an "rm -r" on the venv_path if it exists and nothing otherwise.

evansde77 commented:
LGTM

jcounts commented on src/cirrus/build.py:
```diff
@@ -45,6 +67,13 @@ def main():
+
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
+        cmd = "rm -r {0}".format(venv_path)
```
maybe `-rf`?

ksnavely replied:
The `f` is only needed if you do not have permissions to change the file, which you should have in your local dev environment and I believe we'll have in the CI environment.

ksnavely replied:
Woops, that's wrong. Lesson: Do not review code at 10:30 at night. You may indeed want the f option.

shudgston replied:
You may want to try out https://docs.python.org/2/library/shutil.html#shutil.rmtree, just to remove os-specific commands.

shudgston replied:
Maybe you are fine.  Didn't see the fabric stuff until now.  I'll let others comment on this.

ksnavely replied:
A good thing to keep in mind in general; the scope of cirrus is probably for *nix flavored OS's. Pragmatically this is OS X and Linux.

jcounts commented:
Small comment but +1 overall. Convenient little addition :-)

ksnavely commented on src/cirrus/build.py:
```diff
@@ -11,21 +11,43 @@
+    parser = ArgumentParser(
+        description = 'git cirrus build'
+    )
+    parser.add_argument('command',nargs='?')
```
Make sure you are either running pep8 or have a pep8 plugin running in your editor (space after comma)

ksnavely commented on src/cirrus/build.py:
```diff
@@ -11,21 +11,43 @@
+
+    """
+    parser = ArgumentParser(
+        description = 'git cirrus build'
```
pep8

ksnavely commented on src/cirrus/build.py:
```diff
@@ -11,21 +11,43 @@
       - defaults to ./requirements.txt for reqs
+    - removes existing virtualenv if clean flag is set
     - builds the virtualenv
     - pip installs the requirements into it
```
Could you document `opts` and its type? We typically use sphinx style `:param FooType opts: A FooType of blah`

ksnavely commented on src/cirrus/build.py:
```diff
@@ -45,6 +67,13 @@ def main():
         'virtualenv')
+
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
```
`if opts.clean` is probably more idiomatic than `if opts.clean is True`

shudgston replied:
It is

ksnavely commented on src/cirrus/build.py:
```diff
@@ -45,6 +67,13 @@ def main():
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
+        cmd = "rm -r {0}".format(venv_path)
+        print "Removing existing virtualenv: {0}".format(venv_path)
```
You should use the logger. https://github.com/evansde77/cirrus/blob/b13e908690c1c58b244ce1269f1bfccc62f052ae/src/cirrus/deploy.py#L18

In general one should avoid print (if logging facilities are provided); if you do need to use it then use the functional form `print(...)`.

ksnavely commented on src/cirrus/build.py:
```diff
@@ -11,21 +11,43 @@
 
 
-def main():
+def build_parser(argslist):
```
Document argslist in the docstring?

ksnavely commented:
+1 with some minor nits

appeltel commented:
Ok, thanks for the corrections. I added a force flag to rm, but kept with fabric for consistency, added shpinx-style parameters to the docstrings, actually ran pep8 (oops!), removed the "is True", and made the module use the logger.
